### PR TITLE
122 create web profile page

### LINKED ISSUE #122: Create web profile page
## Describe the feature
Create profile page for web

## TO DO
Create a button on home page to navigate to the profile page

## Extra links (Optional)
_Add any other extra links_

## Screenshots (Optional)
_Add any screenshot_

### PR DESCRIPTION
**Pull request by osvegn**
# New feature
Create settings page with deconnexion button
Add user information in cache to store the session authentication
image

### CODE REVIEW

File: Web/src/AuthPage.js [hunk 1]
```diff
@@ -9,12 +9,14 @@ import "./App.css"
...
 
+import { getAllCacheData, addDataIntoCache } from './CacheManagement'
+
 /**
  * @brief It creates the Sign up and Sign In Pages for the AREA
  * It has a login with facebook on Sign In page
  * @returns the page corresponding to the current authMode
  */
-export default function AuthPage() {
+export default function AuthPage(props) {
```
Comment (31Nathan) [inline Web/src/AuthPage.js]: Missing Props documentation

File: Web/src/CacheManagement.js [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+    const data = new Response(JSON.stringify(response));
+    if ('caches' in window) {
+        caches.open(cacheName).then((cache) => {
+            console.log("cache: " + JSON.stringify(response));
+            cache.put(url, data);
+        });
+    }
+};
+
+// export const cache = () => {
```
Comment (31Nathan) [inline Web/src/CacheManagement.js]: Why do you save this code as commented ?

File: Web/src/CacheManagement.js [hunk 3]
```diff
@@ -0,0 +1,53 @@
+import { ip } from './env'
+
+export async function getAllCacheData() {
```
Comment (31Nathan) [inline Web/src/CacheManagement.js:7]: Documentation

File: Web/src/SettingsPage.js [hunk 4]
```diff
@@ -0,0 +1,192 @@
...
+import HomeImage from './assets/logo.png';
+import GoogleImage from './assets/google.png';
+import SpotifyImage from './assets/spotify.png';
+import TwitterImage from './assets/twitter.png';
+import TwitchImage from './assets/twitch.png';
+import StravaImage from './assets/strava.png';
+import DeconnexionImage from './assets/deconnexion.png';
+import { getAllCacheData, addDataIntoCache } from './CacheManagement'
+
+export default function SettingsPage(props) {
```
Comment (31Nathan) [inline Web/src/SettingsPage.js:168]: Documentation

**Review by 31Nathan — CHANGES_REQUESTED** [review]
Documentation mostly

**Review by 31Nathan — DISMISSED** [review]
LGTM

**Review by 31Nathan — DISMISSED** [review]
LGTM

**Review by CedricCORGE — DISMISSED** [review]
NTR

**Review by STom6 — DISMISSED** [review]
lgtm

**Review by 31Nathan — APPROVED** [review]
LGTM

**Review by CedricCORGE — APPROVED** [review]
lgtm

**Pull request merged**